Don't skip register_dataclass unflatten in dynamo

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #163137
* #163136
* #163107
* #162993
* #162992
* #162682
* #162559
* #162558
* __->__ #162557


We changed how we are tracing, as a result, we need to trace into register_data_class now.

Differential Revision: [D82478651](https://our.internmc.facebook.com/intern/diff/D82478651)

tugsbayasgalan committed:
Don't skip register_dataclass unflatten in dynamo

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Don't skip register_dataclass unflatten in dynamo"




[ghstack-poisoned]

tugsbayasgalan commented on test/export/test_swap.py:
```diff
@@ -377,12 +377,15 @@ class Foo(torch.nn.Module):
             def forward(self, x, *, inputs):
                 return x + torch.matmul(inputs.a, inputs.b)
 
```
Sure.

tugsbayasgalan committed:
Update on "Don't skip register_dataclass unflatten in dynamo"



We changed how we are tracing, as a result, we need to trace into register_data_class now. 


[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Don't skip register_dataclass unflatten in dynamo"



We changed how we are tracing, as a result, we need to trace into register_data_class now. 


[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Don't skip register_dataclass unflatten in dynamo"



We changed how we are tracing, as a result, we need to trace into register_data_class now. 
Differential Revision: D82180555


[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Don't skip register_dataclass unflatten in dynamo"



We changed how we are tracing, as a result, we need to trace into register_data_class now. 
Differential Revision: D82180555


[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Don't skip register_dataclass unflatten in dynamo"



We changed how we are tracing, as a result, we need to trace into register_data_class now.

Differential Revision: [D82478651](https://our.internmc.facebook.com/intern/diff/D82478651)

[ghstack-poisoned]

angelayi commented on torch/_export/utils.py:
```diff
@@ -488,6 +488,7 @@ def register_dataclass_as_pytree_node(
         f"Only dataclasses can be registered with this function: {cls}"
     )
 
+    @torch._dynamo.dont_skip_tracing
```
do we still need this then?

angelayi commented on test/export/test_swap.py:
```diff
@@ -377,12 +377,15 @@ class Foo(torch.nn.Module):
+                {"inputs": CustomInput(torch.randn(2, 3), torch.randn(3, 2))},
+                strict=self.strict,
+                pre_dispatch=True,
+                _use_new_tracer_experimental=use_new_tracer,
```
why cant this just call export?

tugsbayasgalan replied:
oh i think i was just debugging and forgot to change it back.

tugsbayasgalan committed:
Update on "Don't skip register_dataclass unflatten in dynamo"



We changed how we are tracing, as a result, we need to trace into register_data_class now.

Differential Revision: [D82478651](https://our.internmc.facebook.com/intern/diff/D82478651)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Don't skip register_dataclass unflatten in dynamo"



We changed how we are tracing, as a result, we need to trace into register_data_class now.

Differential Revision: [D82478651](https://our.internmc.facebook.com/intern/diff/D82478651)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D82478651).

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D82478651).

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D82478651).

tugsbayasgalan commented:
@pytorchbot merge -f "Landed internally"
